tf2/src/time.cpp

```diff
@@ -44,30 +43,18 @@ std::string tf2::displayTimePoint(const TimePoint& stamp)
   double current_time = timeToSec(stamp);
   int buff_size = snprintf(NULL, 0, format_str, current_time);
   if (buff_size < 0) {
-#ifdef _WIN32
-    // Using fixed buffer size since, strerrorlen_s not yet available
-    const int errormsglen = 200;
-    char errmsg[errormsglen];
-    strerror_s(errmsg, errormsglen, errno);
+    char errmsg[200];
+    rcutils_safe_strerror(errmsg, sizeof(errmsg));
     throw std::runtime_error(errmsg);
-#else
-    throw std::runtime_error(strerror(errno));
-#endif // _WIN32
   }
 
   char * buffer = new char[buff_size];
   int bytes_written = snprintf(buffer, buff_size, format_str, current_time);
   if (bytes_written < 0) {
     delete[] buffer;
-#ifdef _WIN32
-    // Using fixed buffer size since, strerrorlen_s not yet available
-    const int errormsglen = 200;
-    char errmsg[errormsglen];
-    strerror_s(errmsg, errormsglen, errno);
+    char errmsg[200];
+    rcutils_safe_strerror(errmsg, sizeof(errmsg));
     throw std::runtime_error(errmsg);
-#else
-    throw std::runtime_error(strerror(errno));
-#endif // _WIN32
```

Comment:
Instead, replace all of this with `rcutils_snprintf()`?

Reply:
For consistencies sake, I did switch to `rcutils_snprintf`.  As far as I can tell, this doesn't remove the need for the rest of the error checking, so I left the rest as-is.

Reply:
Oh maybe I meant to say format string, it does the whole “how big? Allocate, snprintf, error handling” thing:

https://github.com/ros2/rcutils/blob/bd590e5253672957e8c7e070523fa725a4187db2/src/format_string.c#L32

Though it would use an rcutils allocator instead. 

Idiomatic c++ would use streams instead... or the format stuff if we had c++20.